presto-spi/src/main/java/io/prestosql/spi/expression/ConnectorExpression.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.expression;
+
+import io.prestosql.spi.type.Type;
+
+public class ConnectorExpression
```

Comment:
why is this class needed? It only wraps `type` object

Reply:
It should be abstract. It’s the top of the expression hierarchy

Reply:
Currently only VariableReference is supported. As we keep expanding it, we'll support invocations, constants, etc., which will allow us to model any expression.